### PR TITLE
test: ism_multisig

### PR DESCRIPTION
**Pull request by hashableric**
* I apologize this PR being huge.
* I applied tests to all executable / queryable function in hpl-ism-multisig 
* closes MITO-17

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ cw-storage-plus = "1.0.1"
 cw-utils = "0.16.0"
 cw2 = "1.0.0"
 sha3 = "0.10.8"
+sha2 = { version = "0.10.6", default-features = false }
```
Comment (byeongsu-hong) [inline Cargo.toml:33]:
```suggestion
sha2 = { version = "0.10.6", default-features = false }
secp256k1 = "0.27.0"
```

File: contracts/ism-multisig/Cargo.toml [hunk 2]
```diff
@@ -32,6 +32,7 @@ thiserror = { workspace = true }
 cosmwasm-schema = { workspace = true }
 
 hpl-interface = { workspace = true }
+secp256k1 = "0.27.0"
```
Comment (byeongsu-hong) [inline contracts/ism-multisig/Cargo.toml]:
```suggestion
secp256k1 = { workspace = true }
```
...and can you move this line to above? it'll be good to specify the other workspace packages

File: contracts/ism-multisig/src/contract.rs [hunk 3]
```diff
@@ -75,51 +70,14 @@ pub fn execute(
...
-            let success = unique_vali_pubkey
-                .intersection(&unique_meta_pubkey)
-                .collect::<Vec<_>>()
-                .len();
-
-            Ok(to_binary(&VerifyResponse {
-                verified: success >= usize::from(threshold),
-            })?)
-        }
+        } => query::get_verify(deps, raw_metadata, raw_message),
```
Comment (byeongsu-hong) [inline contracts/ism-multisig/src/contract.rs]: how about using verify_message?

File: contracts/ism-multisig/src/error.rs [hunk 4]
```diff
@@ -13,8 +13,23 @@ pub enum ContractError {
     Unauthorized {},
 
     #[error("Wrong length")]
-    WrongLength,
+    WrongLength {},
```
Comment (byeongsu-hong) [inline contracts/ism-multisig/src/error.rs]: It is redundant to use `{}` to enum that does not contains any field

File: packages/interface/src/types/message.rs [hunk 5]
```diff
@@ -74,7 +74,7 @@ mod tests {
 
     #[test]
     fn test_encode_decode() {
-        let encode_expected = HexBinary::from_hex("00000021500000aef3000000000000000000000000477d860f8f41bc69ddd32821f2bf2c2af0243f1600aa36a70000000000000000000000005d56b8a669f50193b54319442c6eee5edd66238148656c6c6f21").unwrap();
+        let encode_expected: HexBinary = HexBinary::from_hex("00000021500000aef3000000000000000000000000477d860f8f41bc69ddd32821f2bf2c2af0243f1600aa36a70000000000000000000000005d56b8a669f50193b54319442c6eee5edd66238148656c6c6f21").unwrap();
```
Comment (byeongsu-hong) [inline packages/interface/src/types/message.rs]: redundant type declaration

**Review by byeongsu-hong — CHANGES_REQUESTED** [review]

**Comment by byeongsu-hong** [comment]
Can you declare the issue like this? closes MITO-17

File: packages/interface/src/types/message.rs (hunk 5 above)
Comment (hashableric) [reply packages/interface/src/types/message.rs]: i didnt realize i fix it :(

File: contracts/ism-multisig/Cargo.toml [hunk 6]
```diff
@@ -32,7 +32,7 @@ thiserror = { workspace = true }
 cosmwasm-schema = { workspace = true }
 
 hpl-interface = { workspace = true }
-secp256k1 = "0.27.0"
+secp256k1 = { workspace = true }
```
Comment (byeongsu-hong) [inline contracts/ism-multisig/Cargo.toml:33]:
```suggestion
```

File: contracts/ism-multisig/Cargo.toml [hunk 7]
```diff
@@ -32,7 +32,7 @@ thiserror = { workspace = true }
 cosmwasm-schema = { workspace = true }
```
Comment (byeongsu-hong) [inline contracts/ism-multisig/Cargo.toml:32]:
```suggestion
cosmwasm-schema = { workspace = true }
secp256k1 = { workspace = true }
```

**Review by byeongsu-hong — CHANGES_REQUESTED** [review]

**Review by steve806 — APPROVED** [review]

**Review by byeongsu-hong — APPROVED** [review]

**Pull request merged**